controllers/controllers_test.go

```diff
@@ -12,39 +12,39 @@ func TestGetAddress(t *testing.T) {
 	testCases := []struct {
 		name        string
 		input       []core.NodeAddress
-		expectedOut []string
+		expectedOut string
 		expectedErr bool
 	}{
 		{
 			name:        "no addresses",
 			input:       []core.NodeAddress{{}},
-			expectedOut: []string{""},
+			expectedOut: "",
 			expectedErr: true,
 		},
 		{
 			name:        "ipv6",
 			input:       []core.NodeAddress{{Type: core.NodeInternalIP, Address: "::1"}},
-			expectedOut: []string{""},
+			expectedOut: "",
 			expectedErr: true,
 		},
 		{
 			name:        "ipv4",
 			input:       []core.NodeAddress{{Type: core.NodeInternalIP, Address: "127.0.0.1"}},
-			expectedOut: []string{"127.0.0.1"},
+			expectedOut: "127.0.0.1",
 			expectedErr: false,
 		},
 		{
 			name:        "dns",
 			input:       []core.NodeAddress{{Type: core.NodeInternalDNS, Address: "localhost"}},
-			expectedOut: []string{"localhost"},
+			expectedOut: "localhost",
 			expectedErr: false,
 		},
 		{
 			name: "dns and ipv4",
 			input: []core.NodeAddress{
 				{Type: core.NodeInternalDNS, Address: "localhost"},
 				{Type: core.NodeInternalIP, Address: "127.0.0.1"}},
-			expectedOut: []string{"localhost", "127.0.0.1"},
+			expectedOut: "localhost",
```

Comment:
Consider moving the `NodeInternalDNS` address to the bottom of the array to ensure we pick the preferred one.